ibc-core/ics24-host/types/src/error.rs

```diff
@@ -9,6 +9,51 @@ use ibc_primitives::prelude::*;
 use ibc_primitives::proto::Error as ProtoError;
 use prost::DecodeError as ProstError;
 
+/// Errors that originate from host implementations.
+#[derive(Debug, Display)]
+pub enum HostError {
+    /// invalid state: `{description}`
+    InvalidState { description: String },
+    /// missing state: `{description}`
+    MissingState { description: String },
+    /// failed to update store: `{description}`
+    FailedToStore { description: String },
+    /// failed to retrieve from store: `{description}`
+    FailedToRetrieve { description: String },
+    /// failed to parse data: `{description}`
+    FailedToParse { description: String },
+    /// failed to validate client: `{description}`
+    FailedToValidateClient { description: String },
```

Comment:
I am even thinking this can be merged with the `InvalidState`, specifically the description field will be containing a client-specific error, which should sufficient. What do you think?

Reply:
[c8ad6e5](https://github.com/cosmos/ibc-rs/pull/1331/commits/c8ad6e5156eb7b09b3676fa46902cd917473b819)